subprojects/core/src/main/java/org/gradle/api/services/internal/BuildServiceProvider.java

```diff
@@ -16,24 +16,19 @@
 
 package org.gradle.api.services.internal;
 
```

Comment:
Refactored BuildServiceProvider into two subclasses: RegisteredBSP and ConsumedBSP